### PR TITLE
Better example

### PR DESCRIPTION
**Pull request by kosta**
[Rendered Readme](https://github.com/understand-ai/async-transformers/blob/better-example/Readme.md)

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -16,42 +16,89 @@ Async-Transformers is a tiny no-frills no-dependencies ts-first implementation o
...
 
-## Usage
+## Example
+
+Here's an example that downloads all status code images from [http.cat](https://http.cat), but only 7 at a time to be a good
+internet citizen.
+
+You can run this using `npx ts-node examples/fetch-http-cats.ts`
+
+(Note: this library does _not_ depend on nodejs (and has zero dependencies), just this example)
```
Comment (Germandrummer92) [inline Readme.md:28]:
```suggestion
(Note: this library does _not_ depend on node-fetch (and has zero dependencies), just this example does)
```

File: examples/fetch-http-cats.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { PromiseWrapper, asyncBufferedTransformer } from "../dist";
```
Comment (Germandrummer92) [inline examples/fetch-http-cats.ts:1]: hmm this import seems weird from dist?

File: package.json [hunk 3]
```diff
@@ -49,4 +51,4 @@
   "publishConfig": {
     "access": "public"
   }
-}
\ No newline at end of file
+}
```
Comment (Germandrummer92) [inline package.json:54]: I really think you should fix your vscode so that it doesn't delete newlines and makes these files ILLEGAL.

File: Readme.md [hunk 4]
```diff
@@ -16,42 +16,89 @@ Async-Transformers is a tiny no-frills no-dependencies ts-first implementation o
...
-        const nextOutput = i;
-        yield {
-            promise: new Promise((resolve) =>
-                // do the expensive processing / io-requests, can of course also deal with an async function
-                resolve(nextOutput)
-            )
-        }
-        i++;
-    }
+import { PromiseWrapper, asyncBufferedTransformer } from "../dist";
```
Comment (Germandrummer92) [inline Readme.md:31]: this also feels like a weird import

File: Readme.md (hunk 1 above)
Comment (Germandrummer92) [reply Readme.md:28]: thats what you meant, right?

(i guess it could be both, but it was confusing to me either way so maybe rephrase it either way)

**Review by Germandrummer92 — CHANGES_REQUESTED** [review]
tiny comments

File: examples/fetch-http-cats.ts (hunk 2 above)
Comment (kosta) [reply examples/fetch-http-cats.ts:1]: Oops, probably my editor trying to refactor when I moved this into the example folder.

File: Readme.md (hunk 4 above)
Comment (kosta) [reply Readme.md:31]: Fixed.

File: package.json (hunk 3 above)
Comment (kosta) [reply package.json:54]: Is this file illegal if it has an extra linebreak?

Comment (kosta) [reply package.json:54]: I think I added it? In any case, it has a newline now.
